bundles/mapping/mapmodule/MapModuleClass.ol.js

```diff
@@ -538,24 +599,39 @@ export class MapModule extends AbstractMapModule {
      * @param {Boolean} suppressEnd true to NOT send an event about the map move
      *  (other components wont know that the map has moved, only use when chaining moves and
      *     wanting to notify at end of the chain for performance reasons or similar) (optional)
+     * @param {Object} options animation name as string
+     *     Usable animations: fly/pan/zoomPan
      * @return {Boolean} success
      */
-    centerMap (lonlat, zoom, suppressEnd) {
+    centerMap (lonlat, zoom, suppressEnd, options) {
+        const view = this.getMap().getView();
         lonlat = this.normalizeLonLat(lonlat);
         if (!this.isValidLonLat(lonlat.lon, lonlat.lat)) {
             return false;
         }
-        this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
         if (zoom === null || zoom === undefined) {
             zoom = this.getMapZoom();
         }
-        this.getMap().getView().setZoom(zoom);
+
+        const zoomN = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
```

Comment:
We need to check if zoom was a number and treat it as zoom.value and use this logic if zoom was an object to be backwards compatible.